manifests/ssltools/nssdb.pp

```diff
@@ -1,25 +1,13 @@
 # Sets up nssdb
 class certs::ssltools::nssdb (
   $nss_db_dir = $certs::nss_db_dir,
+  $nss_db_password_file = $certs::nss_db_password_file,
   $group = 'qpidd',
 )  {
   Exec { logoutput => 'on_failure' }
 
-  $nss_db_password_file   = "${nss_db_dir}/nss_db_password-file"
-
-  $nssdb_files = $facts['os']['release']['major'] ? {
-    '7' => ["${nss_db_dir}/cert8.db", "${nss_db_dir}/key3.db", "${nss_db_dir}/secmod.db"],
-    default => ["${nss_db_dir}/cert9.db", "${nss_db_dir}/key4.db", "${nss_db_dir}/pkcs11.txt"]
-  }
-
   ensure_packages(['openssl', 'nss-tools'])
 
-  file { $nss_db_dir:
-    ensure => directory,
-    owner  => 'root',
-    group  => $group,
-    mode   => '0755',
-  } ->
   exec { 'generate-nss-password':
```

Comment:
Capturing an IRC discussion point, can we move this to use the `cache_data` model we use everywhere else?

Reply:
Done! See https://github.com/theforeman/puppet-certs/pull/344/commits/6d8af5e6ffe97174f058ad61e3cb6ee7a568f438